spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/jsbridge/SCJavascriptBridgeAPI.java

```diff
@@ -0,0 +1,513 @@
+                            }
+
+                            payload.put("stores", storesArray);
+
+                            subscriber.onNext(payload);
+                        }
+                    }
+                });
+    }
+
+    /**
+     * Handles all the {@link Actions#DATASERVICE_ACTIVESTOREBYID} commands.
+     */
+    private void handleActiveStoreById(HashMap<String, Object> payload, Subscriber<Object> subscriber) {
+        String storeId = JsonUtilities.getString(payload, "storeId");
+        SCDataStore store = mSpatialConnect.getDataService().getStoreByIdentifier(storeId);
+        subscriber.onNext(getStoreMap(store));
+        subscriber.onCompleted();
+    }
+
+    /**
+     * Handles all the {@link Actions#DATASERVICE_STORELIST} commands.
+     */
+    private void handleStoreList(final Subscriber<Object> subscriber) {
+        subscriber.onNext(getAllStoresPayload());
+
+        mSpatialConnect.getDataService().getStoreEvents().subscribe(new Action1<SCStoreStatusEvent>() {
+            @Override
+            public void call(SCStoreStatusEvent scStoreStatusEvent) {
+                subscriber.onNext(getAllStoresPayload());
+            }
+        });
+    }
+
+    /**
+     * Handles the {@link Actions#DATASERVICE_QUERYALL} and
+     * {@link Actions#DATASERVICE_SPATIALQUERYALL} commands.
+     */
+    private void handleQueryAllStores(HashMap<String, Object> payload, final Subscriber<Object> subscriber) {
+        SCQueryFilter filter = getFilter(JsonUtilities.getHashMap(payload, "filter"));
+
+        mSpatialConnect.getDataService().queryAllStores(filter)
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Action1<SCSpatialFeature>() {
+                    @Override
+                    public void call(SCSpatialFeature feature) {
+                        try {
+                            // base64 encode id and set it before sending across wire
+                            String encodedId = feature.getKey().encodedCompositeKey();
+                            feature.setId(encodedId);
+                            subscriber.onNext(feature.toJSON());
+                        } catch (UnsupportedEncodingException e) {
+                            subscriber.onError(e);
+                        }
+                    }
+                });
+    }
+
+    /**
+     * Handles the {@link Actions#DATASERVICE_QUERY} and
+     * {@link Actions#DATASERVICE_SPATIALQUERY} commands.
+     */
+    private void handleQueryStoresByIds(HashMap<String, Object> payload, final Subscriber<Object> subscriber) {
+        SCQueryFilter filter = getFilter(JsonUtilities.getHashMap(payload, "filter"));
+
+        ArrayList<String> storeIds = JsonUtilities.getArrayList(payload, "storeId", String.class);
+
+        mSpatialConnect.getDataService().queryStoresByIds(storeIds, filter)
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Action1<SCSpatialFeature>() {
```

Comment:
I think we need to handle `onError` here. if one of the store querys throws an error, it wouldn't be caught here. See: https://github.com/boundlessgeo/react-native-spatialconnect/blob/930b62b358018287f2c67872534939e641bc1538/android/src/main/java/com/boundlessgeo/spatialconnect/jsbridge/RNSpatialConnect.java#L624